diff a/raddb/all.mk b/raddb/all.mk
--- a/raddb/all.mk
+++ b/raddb/all.mk
@@ -8,7 +8,7 @@ DEFAULT_SITES :=	default inner-tunnel
 LOCAL_SITES :=		$(addprefix raddb/sites-enabled/,$(DEFAULT_SITES))
 
 DEFAULT_MODULES :=	always attr_filter cache_eap chap date \
-			detail detail.log digest dhcp dynamic_clients eap \
+			detail detail.log digest dynamic_clients eap \
 			echo exec expiration expr files linelog logintime \
 			mschap ntlm_auth pap passwd preprocess radutmp realm \
 			replicate soh sradutmp unix unpack utf8
