diff a/raddb/all.mk b/raddb/all.mk
Index: freeradius-server/raddb/all.mk
===================================================================
--- freeradius-server.orig/raddb/all.mk
+++ freeradius-server/raddb/all.mk
@@ -8,7 +8,7 @@ DEFAULT_SITES :=	default inner-tunnel
 LOCAL_SITES :=		$(addprefix raddb/sites-enabled/,$(DEFAULT_SITES))

 DEFAULT_MODULES :=	always attr_filter cache_eap chap client \
-			detail detail.log digest dhcp eap \
+			detail detail.log digest eap \
 			eap_inner echo exec expiration expr files linelog logintime \
 			mschap ntlm_auth pap passwd radutmp realm \
 			replicate soh sradutmp unix unpack utf8
