diff --git a/scripts/logrotate/freeradius b/scripts/logrotate/freeradius
index cbeeb5f..bfb8220 100644
--- a/scripts/logrotate/freeradius
+++ b/scripts/logrotate/freeradius
@@ -1,7 +1,7 @@
 #
 #  Sample logrotate file for FreeRADIUS
 #
-#  You can use this to rotate the /var/log/radius/* files, simply copy it to /etc/logrotate.d/radiusd
+#  You can use this to rotate the /var/log/freeradius/* files, simply copy it to /etc/logrotate.d/radiusd
 #
 
 #
@@ -17,28 +17,28 @@ notifempty
 #
 #  The main server log
 #
-/var/log/radius/radius.log {
+/var/log/freeradius/radius.log {
 	copytruncate
 }
 
 #
 #  Session monitoring utilities
 #
-/var/log/radius/checkrad.log /var/log/radius/radwatch.log {
+/var/log/freeradius/checkrad.log /var/log/freeradius/radwatch.log {
 	nocreate
 }
 
 #
 #  Session database modules
 #
-/var/log/radius/radutmp /var/log/radius/radwtmp {
+/var/log/freeradius/radutmp /var/log/freeradius/radwtmp {
 	nocreate
 }
 
 #
 #  SQL log files
 #
-/var/log/radius/sqllog.sql {
+/var/log/freeradius/sqllog.sql {
 	nocreate
 }
 
@@ -49,6 +49,6 @@ notifempty
 # (or similar) in radiusd.conf, without rotation.  If you go with the
 # second technique, you will need another cron job that removes old
 # detail files.  You do not need to comment out the below for method #2.
-/var/log/radius/radacct/*/detail {
+/var/log/freeradius/radacct/*/detail {
 	nocreate
 }
